word_movers_distance/Calculate_WMD_cos_scores.ipynb

```diff
@@ -0,0 +1,3817 @@
+       "14  0.601292                    cultural                  relational  \n",
+       "15  0.597985                  relational                  relational  \n",
+       "16  0.555859                    cultural                  relational  \n",
+       "17  0.502839                    cultural                  relational  \n",
+       "18  0.599960                    cultural                    cultural  \n",
+       "19  0.595760                    cultural                  relational  \n",
+       "\n",
+       "[20 rows x 23 columns]"
+      ]
+     },
+     "execution_count": 73,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pos_demog"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 77,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "drop_columns= ['Unnamed: 0','Unnamed: 0.1','text','mean_wmd_score', 'top10_sims_articles','text_no_tags','processed']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 82,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos_cult = pos_cult.drop(drop_columns,axis = 1 )\n",
+    "neg_cult = neg_cult.drop(drop_columns,axis = 1 )\n",
+    "pos_demog = pos_demog.drop(drop_columns,axis = 1 )\n",
+    "neg_demog = neg_demog.drop(drop_columns,axis = 1 )\n",
+    "pos_rela = pos_rela.drop(drop_columns,axis = 1 )\n",
+    "neg_rela = neg_rela.drop(drop_columns,axis = 1 )\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 84,
+   "metadata": {
+    "id": "K8SjK1HfiTdR"
+   },
+   "outputs": [],
+   "source": [
+    "pos_cult.to_csv('pos_cult_sample_wmd_cos.csv')\n",
+    "neg_cult.to_csv('neg_cult_sample_wmd_cos.csv')\n",
+    "pos_demog.to_csv('pos_demog_sample_wmd_cos.csv')\n",
+    "neg_demog.to_csv('neg_demog_sample_wmd_cos.csv')\n",
+    "pos_rela.to_csv('pos_rela_sample_wmd_cos.csv')\n",
+    "neg_rela.to_csv('neg_rela_sample_wmd_cos.csv')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "7vb2LBlrm6wZ"
+   },
+   "source": [
+    "## Inspecting Distribution of Similarity Metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "metadata": {
+    "id": "ucRNAqP7m_Rn"
+   },
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "metadata": {
+    "id": "JYWDhjaspWHP"
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.preprocessing import normalize"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f015otaRtC8J"
+   },
+   "source": [
+    "#### Overlayed Distributions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "_kNFbir3qxtR"
+   },
+   "source": [
+    "- normalize cosine similarity (range 0-1) and wmd (range 4-6) since they are on different scales\n",
+    "- invert the normalized wmd score, since originally the bigger the wmd, the more distant 2 vectors are"
```

Comment:
Not sure this is the right way normalize and get on the same scale. Instead, I would suggest creating a new column with normalized WMD by subtracting the overall min() from all WMD scores and then dividing all by the max. And instead of making both similarity measures by changing WMD, instead make both DISTANCE measures by subtracting cosine similarity from 1.

Reply:
Has been addressed by implementing this function:
```python
def normalize_invert_wmd(df):
    """
    normalize the wmd, then invert it (1- normalized wmd),
    also normalize cosine simiarlity score.
    
    """
    dem_wmd=df.dem_wmd
    rela_wmd=df.rela_wmd
    cult_wmd=df.cult_wmd
    
    dem_cos = df.dem_cos_sim
    rela_cos = df.rela_cos_sim
    cult_cos = df.cult_cos_sim
    
    normalized_dem_wmd,normalized_rela_wmd,normalized_cult_wmd = [],[],[]
    normalized_dem_cos,normalized_rela_cos,normalized_cult_cos = [],[],[]
    
    for i in np.arange(len(df)):
        normalized_inverted_3_scores_wmd = 1 - normalize([[dem_wmd[i],rela_wmd[i], cult_wmd[i]]])
#         print(normalized_inverted_3_scores)
        normalized_dem_wmd.append(normalized_inverted_3_scores_wmd[0][0])
        normalized_rela_wmd.append(normalized_inverted_3_scores_wmd[0][1])
        normalized_cult_wmd.append(normalized_inverted_3_scores_wmd[0][2])
        
        normalized_inverted_3_scores_cos = normalize([[dem_cos[i],rela_cos[i], cult_cos[i]]])
        
        normalized_dem_cos.append(normalized_inverted_3_scores_cos[0][0])
        normalized_rela_cos.append(normalized_inverted_3_scores_cos[0][1])
        normalized_cult_cos.append(normalized_inverted_3_scores_cos[0][2])
        
    df['normalized_dem_wmd']=normalized_dem_wmd
    df['normalized_rela_wmd']=normalized_rela_wmd
    df['normalized_cult_wmd']=normalized_cult_wmd
    
    df['normalized_dem_cos']=normalized_dem_cos
    df['normalized_rela_cos']=normalized_rela_cos
    df['normalized_cult_cos']=normalized_cult_cos
    
    return df
```

Note that WMD in this implementation is now WORD MOVER'S SIMILARITY (not ideal but OK for testing purposes)